components/motor/gpio/basic.go

```diff
@@ -194,10 +199,11 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 // indicates direction.
 func (m *Motor) SetPower(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
 	m.opMgr.CancelRunning(ctx)
-
 	if math.Abs(powerPct) <= 0.01 {
 		return m.Stop(ctx, extra)
 	}
+	m.mu.Lock()
```

Comment:
Why lock the mutex down here and not at the top? Specifically, suppose this function runs `m.opMgr.CancelRunning()` around the same time that something else calls `GoFor` (which calls `m.opMgr.NewTimedWaitOp`): could that lead to trouble?

Reply:
i wonder this too while reviewing -- stop must be called outside of a mutex to avoid double-locking.